app/assets/javascripts/directives/views/historyMenu.ts

```diff
@@ -1,42 +1,59 @@
-import { WebDirective } from './../../types';
+import { WebDirective } from '../../types';
 import { WebApplication } from '@/ui_models/application';
-import template from '%/directives/session-history-menu.pug';
+import template from '%/directives/history-menu.pug';
 import { SNItem, ItemHistoryEntry, ItemHistory } from '@node_modules/snjs/dist/@types';
+import { PureViewCtrl } from '@/views';
 
-interface SessionHistoryScope {
+interface HistoryScope {
   application: WebApplication
   item: SNItem
 }
 
-class SessionHistoryMenuCtrl implements SessionHistoryScope {
+class HistoryMenuCtrl extends PureViewCtrl implements HistoryScope {
 
-  $timeout: ng.ITimeoutService
   diskEnabled = false
   autoOptimize = false
   application!: WebApplication
   item!: SNItem
-  entries!: ItemHistoryEntry[]
-  history!: ItemHistory
+  sessionHistory?: ItemHistory
+  serverHistory?: ItemHistory
 
   /* @ngInject */
   constructor(
     $timeout: ng.ITimeoutService
   ) {
-    this.$timeout = $timeout;
+    super($timeout);
+    this.state = {
+      fetchingServerHistory: false
+    };
   }
   
   $onInit() {
+    super.$onInit();
     this.reloadHistory();
+    this.fetchServerHistory();
     this.diskEnabled = this.application.historyManager!.isDiskEnabled();
     this.autoOptimize = this.application.historyManager!.isAutoOptimizeEnabled();
   }
 
   reloadHistory() {
-    const history = this.application.historyManager!.historyForItem(this.item);
-    this.entries = history.entries.slice(0).sort((a, b) => {
-      return a.payload.updated_at! < b.payload.updated_at! ? 1 : -1;
+    this.sessionHistory = this.application.historyManager!.sessionHistoryForItem(this.item);
+  }
+
+  get isFetchingServerHistory() {
+    return this.state.fetchingServerHistory;
+  }
+
+  async fetchServerHistory() {
+    this.setState({
+      fetchingServerHistory: true
     });
-    this.history = history;
+    this.serverHistory = await this.application.historyManager!.serverHistoryForItem(this.item)
+      .finally(() => {
+        this.setState({
+          fetchingServerHistory: false
+        });
```

Comment:
If we're using promise functions as control flow statements it'd be more consistent with the rest of the codebase to use built-in language constructs, i-e:

```js
try {
  this.serverHistory = await this.application.historyManager!.serverHistoryForItem(this.item)
} finally {
  this.setState({
    fetchingServerHistory: false
  });
}
```